### PR TITLE
Add Notifications management settings (API 26+)

### PR DESCRIPTION
**Pull request by Sparow199**
Open system notifications setting of Calendula from Notification settings.

### CODE REVIEW

File: Calendula/src/main/res/values/strings_translatable.xml [hunk 1]
```diff
@@ -386,6 +386,8 @@ You will see all their intakes in the home screen."</string>
 <string name="pref_notification_vibration_alarm_audible">"When alarm is audible only"</string>
 <string name="pref_notification_vibration_alarm_silent">"When alarm is silent only"</string>
 <string name="pref_notification_vibration_never">"Never"</string>
+<string name="pref_notification_management">"Manage notifications"</string>
+<string name="pref_notification_management_sum">"Urgency, sound and vibration settings"</string>
```
[the remark this answers is not in the document]
Comment (Sparow199) [reply Calendula/src/main/res/values/strings_translatable.xml]: It's the same text as Facebook Workplace application. But i can modify it.

File: Calendula/src/main/java/es/usc/citius/servando/calendula/settings/notifications/NotificationPrefsContract.kt [hunk 2]
```diff
@@ -30,6 +30,7 @@ interface NotificationPrefsContract {
         fun requestRingtone(reqCode: Int, ringtoneType: Int, currentValue: Uri?)
         fun setNotificationRingtoneText(text: String)
         fun setInsistentRingtoneText(text: String)
+        fun hideNotificationManagementPref()
```
[the remark this answers is not in the document]
Comment (Sparow199) [reply Calendula/src/main/java/es/usc/citius/servando/calendula/settings/notifications/NotificationPrefsContract.kt]: I'll change the code to display application notification settings with all channels.

File: Calendula/src/main/java/es/usc/citius/servando/calendula/settings/notifications/NotificationPrefsPresenter.kt [hunk 3]
```diff
@@ -47,6 +48,8 @@ class NotificationPrefsPresenter(
             view.hideStockPref()
         }
         updateRingtoneSummaries()
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) else { view.hideNotificationManagementPref() }
```
[the remark this answers is not in the document]
Comment (Sparow199) [reply Calendula/src/main/java/es/usc/citius/servando/calendula/settings/notifications/NotificationPrefsPresenter.kt]: I'm agree with you.

File: Calendula/src/main/java/es/usc/citius/servando/calendula/settings/notifications/NotificationPrefsFragment.kt [hunk 4]
```diff
@@ -67,6 +70,16 @@ class NotificationPrefsFragment :
                 presenter.selectInsistentRingtone()
                 return true
             }
+            PreferenceKeys.SETTINGS_NOTIFICATION_MANAGEMENT.key() -> {
+               if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                   val intent = Intent(Settings.ACTION_CHANNEL_NOTIFICATION_SETTINGS).apply {
+                        putExtra(Settings.EXTRA_APP_PACKAGE, applicationPackageName)
+                        putExtra(Settings.EXTRA_CHANNEL_ID, NotificationHelper.CHANNEL_DEFAULT_ID)
```
[the remark this answers is not in the document]
Comment (Sparow199) [reply Calendula/src/main/java/es/usc/citius/servando/calendula/settings/notifications/NotificationPrefsFragment.kt]: I'll change the code in order to display all available channels.

File: Calendula/src/main/res/values/strings_translatable.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ontherunvaro) [reply Calendula/src/main/res/values/strings_translatable.xml]: I was basing my comment on the channel settings screen:

photo_2018-11-05_09-46-22

This is however not anything critical and we can merge even if you don't change it.

[the remark this answers is not in the document]
Comment (Sparow199) [reply Calendula/src/main/res/values/strings_translatable.xml]: I fixed this in my new commit. Thanks

**Comment by Sparow199** [comment]
> This also raises an interesting question: we're hiding the notification channels on builds under O, but we're not hiding the old way on O and upper. We could probably hide the ringtone and vibration prefs on builds over O, since they are managed via the channels.
> 
> That's for another PR though.

I implemented this behavior in my new commit :)

**Comment by ontherunvaro** [comment]
Looks good, thank you for your contribution!

**Pull request merged**